Update RefArch Setup Guide - CORE-491 (#678)
gruntwork-io/docs: _docs-sources/guides/reference-architecture/configuration-guide/index.md

:::

## 1. Create an infrastructure-live repository

1. Create a new repository in your VCS platform. We recommend naming it _infrastructure-live_.
1. Keep this repo handy, as you'll be prompted for the following information in a subsequent step:
    - HTTPS URL (e.g. `https://github.com/gruntwork-io/infrastructure-live`)
    - SSH URL (e.g. `git@github.com:gruntwork-io/infrastructure-live.git`)
    - Default branch (e.g. `main` or `master`)

## 2. Set up the machine user

> The next step is to configure the Machine User Personal Access Token(s)

Reviewer: We switch back and forth between personal access token and PAT in the paragraph and steps below. Should we just define `PAT` up top and switch all other references below to that?

```suggestion
The next step is to configure the Machine User Personal Access Token(s) (PAT)
```

If you are using GitHub to host your `infrastructure-live` repository, you will only need the one 

> personal access token as the permissions will allow access to both your `infrastructure-live` repo and 

Reviewer: ```suggestion
PAT as the permissions will allow access to both your `infrastructure-live` repo and 
```

Gruntwork's private repositories.

If you are using GitLab or Bitbucket to host your `infrastructure-live` repository, you will need a 
Personal Access Token for your respective VCS in addition to a GitHub PAT for access to the 
private Gruntwork GitHub repositories. Note that at this time GitHub is the only supported VCS for 
Reference Architecture deployments.

> First we will create a GitHub Personal Access Token:

Reviewer: Should we define the shorthand here, rather than in the first step below?

```suggestion
First we will create a GitHub Personal Access Token (PAT):
```

> First we will create a GitHub Personal Access Token:

Reviewer: Should we use `PAT` here or fully spell out Personal Access Token? WDYT?

Author: This one felt a little more appropriate to leave full spelled out

- For GitLab, use [these instructions](https://docs.gitlab.com/ee/user/profile/personal_access_tokens.html), and
  grant the following scopes (NOTE: `api`, `read_user`, and `read_api` scopes are only used for uploading the public SSH
  key for the user. You can replace the token with a new one that only has `write_repository` permission after the
  Reference Architecture is deployed.):

  - `write_repository`
  - `api`
  - `read_user`
  - `read_api`

  You should name the token `GitLab-MachineUser-PAT` to be able to identify it later

- For Bitbucket, use [these instructions](https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/), and
  grant the following scopes (NOTE: `Account:Write` is only used for uploading the public SSH key for the user. You can
  replace the token with a new one that only has `Repositories:Write` permission after the Reference Architecture is
  deployed.):

  - `Repositories:Write`
  - `Account:Write`

  You should name the token `BitBucket-MachineUser-PAT` to be able to identify it later.

> Now you will need to create _another_ secret in AWS Secrets Manager in the _shared_ account containing this PAT. You should name the secret following the above naming convention (`GitLab-MachineUser-PAT`/`BitBucket-MachineUser-PAT`). Once the secret is created, **make a note of the ARN**.

Reviewer: ```suggestion
Now you will need to create _another_ secret in AWS Secrets Manager in the _shared_ account containing this PAT. You should name the secret following the above naming convention (`GitLab-MachineUser-PAT`/`BitBucket-MachineUser-PAT`). Once the secret is created, **copy the ARN** to your favorite text editor for later use.
```
Just keeping wording the same between the different VCS providers

1. Setup AWS Organizations in your Root account if you haven't already. Refer to [this
   documentation](https://docs.aws.amazon.com/organizations/latest/userguide/orgs_tutorials_basic.html) for instructions
   on how to setup AWS Organizations.
1. If you do not have one already, [create an IAM
   User](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_users_create.html) with administrator permissions (attach
   the `AdministratorAccess` IAM policy). Make sure to create AWS Access Keys for the IAM User as well.
1. Once you have an IAM User and AWS Access Keys for accessing the Root account, configure your terminal to be able to
   authenticate to the IAM User. If you do not know how to do this, refer to our [Comprehensive Guide to Authenticating
   to AWS on the Command
   Line](https://blog.gruntwork.io/a-comprehensive-guide-to-authenticating-to-aws-on-the-command-line-63656a686799).

## 5. Run the bootstrap script

We're ready to run the wizard to fill in your `reference-architecture-form.yml` with valid values.

Before running the wizard, ensure you have completed steps 1, 2 and 3 and that you have the following values ready at hand:

> - Personal Access Token for YOUR GitHub user. This token is used to create the Pull Request for the Reference Architecture form.

Reviewer: ```suggestion
- PAT for YOUR GitHub user. This token is used to create the Pull Request for the Reference Architecture form.
```

    - If you do not have one, generate a new Personal Access Token with `repo` level permissions.

> - GitHub Machine User Personal Access Token (required in all cases)

Reviewer: ```suggestion
- GitHub Machine User PAT (required in all cases)
```

> - VCS Machine User Personal Access Token (only required if your ultimate infrastructure-live destination is NOT GitHub)

Reviewer: ```suggestion
- VCS Machine User PAT (only required if your ultimate infrastructure-live destination is NOT GitHub)
```

- The HTTPS URL to your VCS `infrastructure-live` repo (e.g., `https://github.com/gruntwork-io/infrastructure-live.git`)
- The SSH URL to your same VCS `infrastructure-live` repo (e.g., `git@github.com:gruntwork-io/infrastructure-live.git`)

In your repo, you will find two scripts:

- `bootstrap_unix.sh`
- `bootstrap_windows.py`

Both scripts will:

1. Sanity check that you have access to the required organizations.
2. Install the Gruntwork command line tool, which does all the heavy lifting for you
3. Run the Gruntwork wizard for you, which helps you:
   - Provision your AWS accounts
   - Register domains
   - Set up your VCS token secrets
   - Fill in your reference-architecture-form.yml file with valid values
   - Commit and push your form to GitHub and open a pull request

Run the corresponding script based on your platform:

### Linux or Mac OS

```bash

> export GITHUB_OAUTH_TOKEN=<YOUR GITHUB PERSONAL ACCESS TOKEN>

Reviewer: ```suggestion
export GITHUB_OAUTH_TOKEN=<YOUR GITHUB PAT>
```

./bootstrap_unix.sh
```

### Windows

Install python, and then run:

```
$env:GITHUB_OAUTH_TOKEN = 'YOUR GITHUB PERSONAL ACCESS TOKEN'
python3 bootstrap_windows.py
```

## 6. Iterate on your form and push your changes up to run your Preflight Checks

![Gruntwork Preflight Checks on GitHub](/img/guides/reference-architecture/configuration-guide/preflight-checks.png)

> Once your form is filled in and pushed to GitHub, our GitHub automations will take over. You'll notice a special GitHub check called _Preflight Checks_ that will run against your `reference-architecture-form.yml` file and flag any errors for you directly in your pull request, like so:

Reviewer: ```suggestion
Once your form is filled in and pushed to GitHub, our automations will take over via Github Actions. You'll notice a special GitHub check called _Preflight Checks_ that will run against your `reference-architecture-form.yml` file and flag any errors for you directly in your pull request, like so:
```
